packages/settings/test/settings-spec.ts

```diff
@@ -1,13 +1,13 @@
 import {
-  resetSettings,
   getSettingsForDocument,
   getDefaultSettings,
   setGlobalSettings,
   setSettingsForDocument,
   hasSettingsForDocument,
   clearDocumentSettings,
   clearSettings,
-} from "../src/settings";
+} from "../src/api";
```

Comment:
Why did you change these imports?

Reply:
importing the `main` of the package.json
causes that file to be executed, thus it gets a coverage report.

Otherwise I got a coverage KPI error.
Regardless it is a good practice to import **Public** APIs the same way a **consumer** would during tests. It in effect creates a bit of both design and runtime "micro test".

Here is how the generated api.**js** looks like:
Note the use of Object.defineProperty (with a **function** in the getter).
So its not enough to load the file, all the exported properties must be **accessed**.

```javascript
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validLoggingLevelValues = exports.validTraceServerValues = void 0;
var settings_1 = require("./settings");
Object.defineProperty(exports, "setGlobalSettings", { enumerable: true, get: function () { return settings_1.setGlobalSettings; } });
Object.defineProperty(exports, "setSettingsForDocument", { enumerable: true, get: function () { return settings_1.setSettingsForDocument; } });
Object.defineProperty(exports, "getSettingsForDocument", { enumerable: true, get: function () { return settings_1.getSettingsForDocument; } });
Object.defineProperty(exports, "hasSettingsForDocument", { enumerable: true, get: function () { return settings_1.hasSettingsForDocument; } });
Object.defineProperty(exports, "clearSettings", { enumerable: true, get: function () { return settings_1.clearSettings; } });
Object.defineProperty(exports, "clearDocumentSettings", { enumerable: true, get: function () { return settings_1.clearDocumentSettings; } });
Object.defineProperty(exports, "getDefaultSettings", { enumerable: true, get: function () { return settings_1.getDefaultSettings; } });
exports.validTraceServerValues = {
    off: true,
    messages: true,
    verbose: true,
};
exports.validLoggingLevelValues = {
    off: true,
    fatal: true,
    error: true,
    warn: true,
    info: true,
    debug: true,
    trace: true,
};
//# sourceMappingURL=api.js.map
```